Create `sphinx` documentation for `simplify`, `join`, `insert`, and `powerset`

hmhummel opened the pull request:
Also, added equivalency of each parameter to Tikka's `causaleffect` functions. Added detail to parameter documentation.

hmhummel committed:
Tidied documentation for simplify. Created documentation for join, insert, and powerset. Still need function name changes and examples.

hmhummel committed:
Rectified changes with function names in simplify documentation

hmhummel committed:
Completed causaleffect to y0 function name changes for join. Still needs unit test/example

hmhummel committed:
Ignored .DS_Store files

hmhummel committed:
Replaced capital letter parameters with more descriptive variable names, and described equivalency of each parametery to `causaleffect` functions. Added more detail to parameter documentation to increase clarity.

djinnome commented on documentation/source/join.rst:
Hey @hmhummel Sphinx will display the mathematical formulas such as $P(J|D)$ and $P(V|C) := P(V_k |C_k)$ in latex if you put a `:math:` before the backquotes, for example:

```
:math:`P(V|C) := P(V_k | C_k)`
```

Note that I included underscores in `Vk` and `Ck` so that they would be subscripted.

hmhummel replied:
Oh, awesome!! Love this & will implement in the next commit — thanks

hmhummel committed:
Added `:math:` before relevant formulas

hmhummel committed:
Replaced capital letter parameters in `insert` with more descriptive variable names, and described equivalency of each parameter to `causaleffect` functions. Added more detail to parameter documentation to increase clarity.

hmhummel committed:
Changed formatting in `simplify` documentation

hmhummel committed:
Replaced capital letter parameters with more descriptive variable names in `simplify` documentation.

hmhummel committed:
Added `causaleffect` package as a reference on main page of documentation

hmhummel committed:
Made formatting changes to `powerset` documentation

hmhummel committed:
Created `unit_tests` directory and a Jupyter notebook for `test_case_1` to begin writing unit tests

hmhummel committed:
Added setup code for `test_case_1`

hmhummel committed:
Ran `sphinx-quickstart` to create html files for `causaleffectpy` documentation

hmhummel committed:
Added prettier theme to sphinx documentation

hmhummel commented:
### Documentation for `causaleffectpy`

I began to document `causaleffectpy` functions in a similar fashion to [`causaleffect`](https://github.com/y0-causal-inference/causaleffect/pull/8) so that this can eventually be integrated into `y0` and used as a reference to write the probabilistic simplification algorithms. 

I used `sphinx` to document 4 functions in `.rst` files, and used [this sphinx tutorial](https://medium.com/@pratikdomadiya123/build-project-documentation-quickly-with-the-sphinx-python-2a9732b66594) with the `make html` command to generate html files.

### Unit testing for `causaleffectpy`
Unit tests for all functions still need to be created with `pytest` using the 3 vignettes/test cases from [`causal.effect` on CRAN](https://cran.r-project.org/web/packages/causaleffect/causaleffect.pdf): 1st with unobserved confounders, 2nd with pruning & fractions, and 3rd with simplest graph possible (no unobserved confounders). For reference, here are the `causal.effect` unit test files using the [`testthat` R package](https://testthat.r-lib.org/): [test_case_1](https://github.com/y0-causal-inference/causaleffect/blob/test-and-document/tests/testthat/test_case_1.R), [test_case_2](https://github.com/y0-causal-inference/causaleffect/blob/test-and-document/tests/testthat/test_case_2.R), and [test_case_3](https://github.com/y0-causal-inference/causaleffect/blob/test-and-document/tests/testthat/test_case_3.R). @djinnome recommended using `causaleffectpy` to help create the unit tests rather than `y0`.

### Needs work:
- [ ] `simplify` - created `sphinx` documentation and html file. Changed names of functions to `y0` but not sure if they're right. Needs an example/unit test
- [ ] `join` - created `sphinx` documentation and html file. Changed names of functions to `y0` but not sure if they're right. Needs an example/unit test
- [ ] `insert` - created `sphinx` documentation and html file. Changed names of functions to `y0` but not sure if they're right. Needs an example/unit test
- [ ] `powerset` - created `sphinx` documentation and html file. No related `y0` functions.
- [ ] `deconstruct`
- [ ] `causal.effect` — might not be needed, as there is already an identification algorithm written for `y0` 
- [ ] `dSep`
- [ ] `wrap.dSep`
- [ ] `parse.expression`
- [ ] `factorize`
- [ ] `probability`
- [ ] `get.expression`
- [ ] etc
